app/src/main/java/woowacourse/movie/MovieAdapter.kt

```diff
@@ -0,0 +1,51 @@
+package woowacourse.movie
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.BaseAdapter
+import android.widget.Button
+import android.widget.ImageView
+import android.widget.TextView
+import woowacourse.movie.model.ScreenMovieUiModel
+
+class MovieAdapter(
+    context: Context,
+    private val movies: List<ScreenMovieUiModel>,
+    private val onClickReservationButton: (id: Long) -> Unit = {},
+) :
+    BaseAdapter() {
+    private val inflater = LayoutInflater.from(context)
+
+    override fun getCount(): Int = movies.size
+
+    override fun getItem(position: Int): ScreenMovieUiModel = movies[position]
+
+    override fun getItemId(position: Int): Long = movies[position].id
+
+    override fun getView(
+        position: Int,
+        convertView: View?,
+        parent: ViewGroup?,
+    ): View {
+        val view = convertView ?: inflater.inflate(R.layout.item_movie, null)
```

Comment:
```suggestion
        val view = LayoutInflater.from(parent?.context).inflate(R.layout.item_movie, null)
```

getView의 `parent: ViewGroup`과  `LayoutInflater` 를 활용하면 외부에서 context를 받지 않아도 가능합니다.

> 1. 외부에서 context를 주입받으면 어떤 단점이 있을지?
> 2. LayoutInflater 는 어떤 역할을 하는지?

위 두가지는 안드로이드를 개발하며 중요한 내용들입니다. 한번 고민해보시고 오둥이의 생각을 공유해주세요 👍

Reply:
오! 이런 방법으로 context 를 가져올 수 있었군요! 👍 👍  
수정하겠습니다 🚀

Reply:
> 1. 외부에서 context를 주입받으면 어떤 단점이 있을지?

메모리 릭의 가능성이 생길 것 같습니다!

>  Activity 보다 생명주기가 긴 객체 A가 Adapter 를 참조한다.
이때,  액티비티가 onDestory 될지라도 A가 살아있으면 context 는 GC에 수거 되지 않습니다.

위와 같은 경우 메모리 릭이 날 수 있을 것 같습니다.  

---

위와 같은 케이스가 아니라면 
activity가 onDestroy 될 때 Adapter 역시 사라지기 때문에  
일반적인 상황에서는 메모리 릭이 일어나지 않을 것이라 생각이 드는데요!   

제이든은 어떻게 생각하시나요??

Reply:
> 2. 	LayoutInflater 는 어떤 역할을 하는지?

XML 파일을 Java로 된 View 객체로 바꾸어 View Tree 의 Node 에 추가해주는 역할을 하는 것으로 알고 있습니다!  

---
item View 의 parent 를 지정하지 않았네요.. 아래와 같이 수정하도록 하겠습니다

```kotlin
// as is
val view = convertView ?: inflater.inflate(R.layout.item_movie, null)

// to be

inflater = LayoutInflater.from(parent?.context)
val view = inflater.inflate(R.layout.item_movie, parent, false)
```